api/users.go

```diff
@@ -516,6 +516,13 @@ func (r usersRoutes) updatePreferredName(c *gin.Context) {
 			return
 		}
 	}
+	if len(request.Value) > 40 {
```

Comment:
We could also add the limit to the database and migrate the existing fields by cutting everything after 40 chars off.

Reply:
(also as in 'in addition')

Reply:
Do you need a hint or want me to implement this? @0ql

Reply:
Do you want me to add the sql rule?

Reply:
That would be great! Also a migration like [these](https://github.com/joschahenningsen/TUM-Live/tree/dev/dao/migrations) might be nice but not required, I don't think anybody did that to their username (except for @tizleo maybe)